ct-aa/dotTheory---TypeScript
# JS-975: Added the sample app for the Node + Express + TypeScript lesson

# Related Tickets
[JS-974](https://codeandtheory.atlassian.net/browse/JS-974)

# Description
- Added the sample app for the Node + Express + TypeScript lesson.

# Validation
- Open the terminal.
- Run cd "node-plus-express-and-typescript" to change the directory.
- Run "npm install" to install the dependencies.
- Run "tsc -w" to watch the typescript files.
- Open another terminal.
- Run "npm start" to start the app.
- The app won't work in the browser. We need to install Postman.
<img width="1387" alt="Screen Shot 2024-01-25 at 5 50 54 PM" src="https://github.com/ct-aa/dotTheory---TypeScript/assets/93116252/9faced92-9331-4e2f-aba0-98fca856e37d">
- Open the Postman app and create a new workspace.
<img width="1736" alt="Screen Shot 2024-01-25 at 5 49 09 PM" src="https://github.com/ct-aa/dotTheory---TypeScript/assets/93116252/75fb41a1-010c-4203-bb2e-2cf001faeacb">
- Create a new test.
<img width="1733" alt="Screen Shot 2024-01-25 at 5 47 03 PM" src="https://github.com/ct-aa/dotTheory---TypeScript/assets/93116252/ef97b069-a6de-4ee2-b5dc-31042a001acc">
- Select "Post"
<img width="1735" alt="Screen Shot 2024-01-25 at 5 53 21 PM" src="https://github.com/ct-aa/dotTheory---TypeScript/assets/93116252/736bd59e-0b81-44ec-8790-9d6e9cadc81b">
- Type "http://localhost:3000/todos/" on the input field.
<img width="1743" alt="Screen Shot 2024-01-25 at 5 54 39 PM" src="https://github.com/ct-aa/dotTheory---TypeScript/assets/93116252/02ca1c23-2401-4aa9-b13e-82a539e99e31">
- Select the following settings and add a sample JSON.
<img width="1737" alt="Screen Shot 2024-01-25 at 5 59 45 PM" src="https://github.com/ct-aa/dotTheory---TypeScript/assets/93116252/bc1132f6-c906-49a4-b22d-62e5933e0cd2">
- Click the Send button
<img width="1733" alt="Screen Shot 2024-01-25 at 6 02 09 PM" src="https://github.com/ct-aa/dotTheory---TypeScript/assets/93116252/496baed2-136c-4f34-a2bd-9f64adf68516">
- You should see this sample response.
<img width="1730" alt="Screen Shot 2024-01-25 at 6 02 41 PM" src="https://github.com/ct-aa/dotTheory---TypeScript/assets/93116252/39b01369-4892-48c4-86b6-b6f11b7bb4d8">

# Risks
N/A
